### PR TITLE
Hide dependent forms until items are not selected in trees

### PR DESCRIPTION
**Pull request by Ruslan-Aleev**
### What does it do?
Hide dependent forms until items are not selected in tree.

You can, for example, set value in a property, try to save, and only then understand that you must first select an element in the tree - which is very confusing.

Before:
forms_hide

After:
forms_hide_2

### Why is it needed?
UI/UX improvement

### Related issue(s)/PR(s)
Logically related to https://github.com/modxcms/revolution/issues/10518

### CODE REVIEW

**Comment by JoshuaLuckers** [comment]
Wouldn't it make more sense to disable the form when nothing is selected so the user understands there is something to edit in the window? If you hide it, you also hide the functionality.

**Comment by Ruslan-Aleev** [comment]
Yes, it makes sense. If you tell me how to do this, I will correct it :) I did the same as in the "User groups" tree.

File: manager/assets/modext/widgets/element/modx.panel.property.set.js [hunk 1]
```diff
@@ -97,6 +97,7 @@ MODx.grid.PropertySetProperties = function(config) {
             ,scope: this
         }]
     });
+    Ext.getCmp('right-column').hide();
```
Comment (JoshuaLuckers) [inline manager/assets/modext/widgets/element/modx.panel.property.set.js]:
```suggestion
    Ext.getCmp('right-column').disable();
```

File: manager/assets/modext/widgets/element/modx.panel.property.set.js [hunk 2]
```diff
@@ -130,6 +132,7 @@ MODx.tree.PropertySets = function(config) {
 };
 Ext.extend(MODx.tree.PropertySets,MODx.tree.Tree,{
     loadGrid: function(n,e) {
+        Ext.getCmp('right-column').show();
```
Comment (JoshuaLuckers) [inline manager/assets/modext/widgets/element/modx.panel.property.set.js]:
```suggestion
        Ext.getCmp('right-column').enable();
```

File: manager/assets/modext/widgets/security/modx.panel.user.js [hunk 3]
```diff
@@ -36,6 +36,7 @@ MODx.panel.User = function(config) {
     MODx.panel.User.superclass.constructor.call(this,config);
     Ext.getCmp('modx-user-panel-newpassword').getEl().dom.style.display = 'none';
     Ext.getCmp('modx-user-password-genmethod-s').on('check',this.showNewPassword,this);
+    Ext.getCmp('modx-extended-form').hide();
```
Comment (JoshuaLuckers) [inline manager/assets/modext/widgets/security/modx.panel.user.js]:
```suggestion
    Ext.getCmp('modx-extended-form').disable();
```

File: manager/assets/modext/widgets/security/modx.panel.user.js [hunk 4]
```diff
@@ -231,7 +232,10 @@ Ext.extend(MODx.panel.User,MODx.FormPanel,{
                         ,prefix: 'extended'
                         ,enableDD: true
                         ,listeners: {
-                            'dragdrop': {fn:function() {
+                            'click': {fn:function() {
+                                Ext.getCmp('modx-extended-form').show();
```
Comment (JoshuaLuckers) [inline manager/assets/modext/widgets/security/modx.panel.user.js]:
```suggestion
                                Ext.getCmp('modx-extended-form').enable();
```

File: manager/assets/modext/widgets/system/modx.panel.dashboard.widget.js [hunk 5]
```diff
@@ -7,6 +7,7 @@
 MODx.panel.DashboardWidget = function(config) {
     config = config || {};
 
+    Ext.getCmp('modx-extended-form').hide();
```
Comment (JoshuaLuckers) [inline manager/assets/modext/widgets/system/modx.panel.dashboard.widget.js]:
```suggestion
    Ext.getCmp('modx-extended-form').disable();
```

File: manager/assets/modext/widgets/system/modx.panel.dashboard.widget.js [hunk 6]
```diff
@@ -216,7 +217,10 @@ MODx.panel.DashboardWidget = function(config) {
                     ,prefix: 'properties'
                     ,enableDD: true
                     ,listeners: {
-                        'dragdrop': {fn:this.markDirty,scope:this}
+                        'click': {fn:function() {
+                            Ext.getCmp('modx-extended-form').show();
```
Comment (JoshuaLuckers) [inline manager/assets/modext/widgets/system/modx.panel.dashboard.widget.js]:
```suggestion
                            Ext.getCmp('modx-extended-form').enable();
```

**Comment by Ruslan-Aleev** [comment]
@JoshuaLuckers That's a lot better. Thanks!

forms_d

**Review by JoshuaLuckers — APPROVED** [review]

**Review by Ibochkarev — APPROVED** [review]

**Review by alroniks — CHANGES_REQUESTED** [review]
On the widget page, it fails with the following error:

```
modx.panel.dashboard.widget.js:10 Uncaught TypeError: Cannot read property 'disable' of undefined
    at new MODx.panel.DashboardWidget (modx.panel.dashboard.widget.js:10)
    at Object.create (ext-all.js:21)
    at MODx.load (modx.js?v=36a40c61:86)
    at MODx.page.UpdateDashboardWidget._loadComponents (modx.component.js:68)
    at MODx.page.UpdateDashboardWidget.MODx.Component [as constructor] (modx.component.js:10)
    at new MODx.page.UpdateDashboardWidget (update.js:36)
    at Object.create (ext-all.js:21)
    at MODx.load (modx.js?v=36a40c61:86)
    at ?a=system/dashboards/widget/update&id=6:27
    at ext-all.js:21
```

**Review by alroniks — APPROVED** [review]

**Comment by alroniks** [comment]
@Ruslan-Aleev Works fine, but what I found and I guess it can be improved in a new PR. It actual for user and dashboard widget code.
When I selected a value it removed disabled flag from the form, but when I selected, for example, container (category) it stays on the previous value, but if it loses focus, maybe it makes sense to remove values from the form, clean it and make disable again.

**Pull request merged**